server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -156,25 +157,7 @@ public static SegmentInfos readSegmentInfos(IndexCommit commit) throws IOExcepti
      * Reads the segments infos from the given segments file name, failing if it fails to load
      */
     private static SegmentInfos readSegmentInfos(String segmentsFileName, Directory directory) throws IOException {
-        // TODO Use readCommit(Directory directory, String segmentFileName, int minSupportedMajorVersion) once Lucene 10.1 is available
-        // and remove the try-catch block for IndexFormatTooOldException
-        assert IndexVersion.current().luceneVersion().equals(LUCENE_10_0_0) : "remove the try-catch block below";
-        try {
-            return SegmentInfos.readCommit(directory, segmentsFileName);
-        } catch (IndexFormatTooOldException e) {
-            try {
-                // Temporary workaround until Lucene 10.1 is available: try to leverage min. read-only compatibility to read the last commit
-                // and then check if this is the commit we want. This should always work for the case we are interested in (archive and
-                // searchable snapshots indices in N-2 version) as no newer commit should be ever written.
-                var segmentInfos = readSegmentInfos(directory);
-                if (segmentsFileName.equals(segmentInfos.getSegmentsFileName())) {
-                    return segmentInfos;
-                }
-            } catch (Exception suppressed) {
-                e.addSuppressed(suppressed);
-            }
-            throw e;
-        }
+        return SegmentInfos.readCommit(directory, segmentsFileName, MIN_SUPPORTED_MAJOR);
```

Comment:
@javanna @tlrx 

In preparation for merging Lucene 10.1, I updated the code here as per commit https://github.com/elastic/elasticsearch/pull/119308/commits/8d729160ce364b4919e19a467a371aaeecc276ef (it's a bit unfortunate that we didn't do this in the lucene_snapshot branch, but can do it now)

Given the relaxation to read N-2 in #118941, and the extensive comments and assertions here, then I think that this change is semantically correct. Can you please confirm, and whether or not an existing constant could be used rather than adding a new one here.